apps/src/aiComponentLibrary/chatMessage/ChatMessage.tsx

```diff
@@ -1,75 +1,106 @@
 import classNames from 'classnames';
-import React from 'react';
+import React, {useMemo, useState} from 'react';
 
+import Button from '@cdo/apps/componentLibrary/button/Button';
 import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
 import {commonI18n} from '@cdo/apps/types/locale';
 import {AiInteractionStatus as Status} from '@cdo/generated-scripts/sharedConstants';
 import aiBotIcon from '@cdo/static/aichat/ai-bot-icon.svg';
 
-import {Role} from './types';
+import {Role, ProfaneMessageViewToggle} from './types';
 
 import moduleStyles from './chat-message.module.scss';
 
-function getDisplayText(chatMessageText: string, status: string, role: Role) {
-  if (status === Status.OK || status === Status.UNKNOWN) {
-    return chatMessageText;
-  }
-
-  if (status === Status.PROFANITY_VIOLATION) {
-    return commonI18n.aiChatInappropriateUserMessage();
-  }
-
-  if (status === Status.PII_VIOLATION) {
-    return commonI18n.aiChatTooPersonalUserMessage();
-  }
-
-  if (status === Status.ERROR) {
-    return role === Role.ASSISTANT
-      ? commonI18n.aiChatResponseError()
-      : chatMessageText;
-  }
-}
-
 interface ChatMessageProps {
   chatMessageText: string;
   role: Role;
   status: string;
+  isTeacherView?: boolean;
 }
 
 const ChatMessage: React.FunctionComponent<ChatMessageProps> = ({
   chatMessageText,
   role,
   status,
+  isTeacherView,
 }) => {
+  const [profaneMessageViewToggle, setProfaneMessageViewToggle] = useState(
+    ProfaneMessageViewToggle.VIEW
+  );
```

Comment:
minor: any reason this can't just be a boolean (`showingProfane/HiddenMessage: true/false`)?